docs/proposals/05-object-storage.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Object Storage
+
+oep-number: 5
+
+creation-date: 2022-12-19
+
+status: implementable
+
+authors:
+
+- @lukasfrank
+- @gehoern
+
+reviewers:
+
+- @adracus
+- @MalteJ
+
+---
+
+# OEP-5: Object Storage
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+Object storage builds the basis for many cloud applications. An Object Storage provides a simplified object 
+model for files, but has a reduced set of security and access features (non-posix). This functionality is build on top 
+of the HTTP protocol. The current market standard is [S3](https://docs.aws.amazon.com/AmazonS3/latest/API/Type_API_Reference.html). 
+This document describes how to integrate simplified S3 buckets into the OnMetal API without taking too many details of 
+the S3 feature completeness itself.
+
+## Motivation
+Object Storage is demanded by cloud native applications, therefore, OnMetal needs to provide it for a complete solution.
+The Object Storage service should be integrated into OnMetal and use all available optimizations. It is not designed 
+to be _just_ a service on top of OnMetal. The used protocol is called [S3](https://docs.aws.amazon.com/AmazonS3/latest/API/Type_API_Reference.html)
+which introduces a storage entity called bucket. For the beginning only the bucket creation and removal is covered.
+
+### Goals
+- S3-compatible object storage implementation
+- Automatic assigned public storage endpoint (addressed via DNS)
```

Comment:
We should not mention DNS here - there is an internet reachable endpoint, that should be enough for the consumer to know.